nativelib/src/main/scala-3/scala/scalanative/runtime/Continuations.scala

```diff
@@ -0,0 +1,168 @@
+package scala.scalanative.runtime
+
+import scala.scalanative.unsafe.*
+import scala.scalanative.runtime.libc.{malloc, free}
+import scala.scalanative.runtime.Intrinsics.*
+import scala.collection.mutable
+
+import scala.util.Try
+import scala.scalanative.meta.LinktimeInfo.isWindows
+
+object Continuations:
+  import ContImpl.*
+
+  opaque type BoundaryLabel[-T] = ContImpl.BoundaryLabel
+
+  ContImpl.init(CFuncPtr2.fromScalaFunction(allocateBlob))
+
+  /** Marks the given body as suspendable with a `BoundaryLabel` that `suspend`
+   *  can refer to. Forwards the return value of `body`, or the return value of
+   *  the `suspend` call, if it happens during the execution of `f`.
+   *
+   *  __Safety__: the passed-in `BoundaryLabel` cannot be used outside of the
+   *  scope of the body. Suspending to a `BoundaryLabel` not created by a
+   *  `boundary` call higher on the same call stack is undefined behaviour.
+   */
+  inline def boundary[T](inline body: BoundaryLabel[T] ?=> T): T =
+    if isWindows then UnsupportedFeature.continuations()
+    val call: ContFnT = (x: ContImpl.BoundaryLabel) =>
+      objToPtr(Try(body(using x)))
+    val resP = ContImpl.boundary(contFn, objToPtr(call))
+    objFromPtr(resP).asInstanceOf[Try[T]].get
+
+  /** Suspends the current running stack up to the corresponding `boundary` into
+   *  a continuation `cont: R => T` and passes it into `f`. The return value of
+   *  `f(cont)` is returned to `boundary`.
+   *
+   *  Same as `suspendCont`, but hides the fact that the passed in function is a
+   *  continuation.
+   */
+  inline def suspend[R, T](
+      inline f: (R => T) => T
+  )(using label: BoundaryLabel[T]): R =
+    suspendCont[R, T](f)
+
+  /** Same as `suspend` where the continuation expects no parameters. */
+  def suspend[T](
+      f: (() => T) => T
+  )(using label: BoundaryLabel[T]): Unit =
+    suspendCont[Unit, T](cont => f(() => cont(())))
+
+  /** Immediately return to the `boundary`. */
+  inline def break(using BoundaryLabel[Unit]): Nothing =
+    suspend[Nothing, Unit](_ => ())
+
+  /** Immediately return to the `boundary`, returning the given value. */
+  inline def break[T](value: T)(using BoundaryLabel[T]): Nothing =
+    suspend[Nothing, T](_ => value)
+
+  inline def suspendCont[R, T](
+      inline f: Continuation[R, T] => T
+  )(using label: BoundaryLabel[T]): R =
+    val cont = Continuation()
+    // We want to reify the post-suspension body here,
+    // while creating the full continuation object.
+    val call = (c: ContImpl.Continuation) =>
+      cont.cont = c
```

Comment:
Just general comment about `cont` - as we, people, tend to read in blocks, and not by invidual characters, it's typically easier to ready full, familiar name, like `continuation` since it already makes a connection in our brain. When we use shortcuts we typically need to analyze and unmangle their meaning. I think that Szymon gave a good talk about this one aspect at Scalar https://www.youtube.com/watch?v=hqvxHHNR8bA&pp=ygURc3p5bW9uIHJvZHppZXdpY3o%3D 
Maybe that's just my personal preference, but `cont` is for me harder to read in code then `continuation` or `suspension` or other real word. 
Anyway that's not a really important issue

Reply:
Got it!